common/persistence/data_manager_interfaces.go

```diff
@@ -2283,8 +2283,23 @@ func (p *TaskListPartition) ToInternalType() *types.TaskListPartition {
 	return &types.TaskListPartition{IsolationGroups: p.IsolationGroups}
 }
 
-// IsActiveActive
+// IsActiveActive returns true if the domain has active-active configuration.
+// Returns true if ClusterAttributes (or legacy RegionToClusters) have been configured for this domain.
 // TODO(active-active): Update unit tests of all components that use this function to cover active-active case
 func (c *DomainReplicationConfig) IsActiveActive() bool {
-	return c != nil && c.ActiveClusters != nil && len(c.ActiveClusters.ActiveClustersByRegion) > 0
+	if c == nil || c.ActiveClusters == nil {
+		return false
+	}
+
+	// Check to see if a ClusterAttribute has been configured for this domain.
+	if len(c.ActiveClusters.AttributeScopes) > 0 {
```

Comment:
this makes me wonder if we really should just make this a distinct attribute on the domain struct. I don't feel strongly, we can come back to it if it's proving annoying